presidio-analyzer/analyzer/recognizer_registry/recognizers_store_api.py

```diff
@@ -0,0 +1,79 @@
+import logging
+import os
+
+import grpc
+import recognizers_store_pb2
+import recognizers_store_pb2_grpc
+
+from analyzer import PatternRecognizer
+from analyzer import Pattern
+
+
+class RecognizerStoreApi:
+    """ The RecognizerStoreApi is the object that talks to the remote
+    recognizers store service and get the recognizers / hash
+    """
+
+    def __init__(self):
+        try:
+            recognizers_store_svc_url = \
+                os.environ["RECOGNIZERS_STORE_SVC_ADDRESS"]
+        except KeyError:
+            recognizers_store_svc_url = "localhost:3004"
+
+        channel = grpc.insecure_channel(recognizers_store_svc_url)
+        self.rs_stub = recognizers_store_pb2_grpc.RecognizersStoreServiceStub(
+            channel)
+
+    def get_latest_hash(self):
+        """
+        Returns the hash of all the stored custom recognizers. Returns empty
+        string in case of an error (e.g. the store is completly empty)
+        """
+        hash_request = \
+            recognizers_store_pb2.RecognizerGetHashRequest()
+
+        last_hash = ""
+        try:
+            last_hash = self.rs_stub.ApplyGetHash(
+                hash_request).recognizersHash
+        except grpc.RpcError:
+            logging.info("Failed to get recognizers hash")
+            return ""
+
+        logging.info("Latest hash found in store is: %d", last_hash)
+        return last_hash
+
+    def get_all_recognizers(self):
+        """
+        Returns a list of CustomRecognizer which were created from the
+        recognizers stored in the underlying store
+        """
+        req = recognizers_store_pb2.RecognizersGetAllRequest()
+        raw_recognizers = []
+
+        try:
+            raw_recognizers = self.rs_stub.ApplyGetAll(req).recognizers
+
+        except grpc.RpcError:
+            logging.info("Failed getting recognizers from the remote store. \
+            Returning an empty list")
+            return raw_recognizers
+
+        custom_recognizers = []
+        for new_recognizer in raw_recognizers:
```

Comment:
Can we use the from_dict method to instantiate a recognizer?

Reply:
I will add it in a later phase